### PR TITLE
Forward solution ID through notifications

### PR DESCRIPTION
**Pull request by sunce86**
# Description
Fixes https://github.com/cowprotocol/services/issues/2002

Now, with each notification, we are sending both auction_id and solution_id which should be enough for all debugging purposes.

Solution ID is represented as `Vec<u64>` since it can contain multiple solutions, as in case of merged settlements.

### CODE REVIEW

File: crates/solvers/src/domain/solution.rs [hunk 1]
```diff
@@ -4,12 +4,31 @@ use {
...
-    std::{collections::HashMap, slice},
+    std::{
+        collections::HashMap,
+        slice,
+        sync::atomic::{AtomicU64, Ordering},
+    },
 };
 
+static COUNTER: AtomicU64 = AtomicU64::new(1);
+pub fn solution_id() -> Id {
```
Comment (MartinquaXD) [inline crates/solvers/src/domain/solution.rs:15]: nit: Now you'd have to use this as `solution::solution_id()`. Maybe it's slightly nicer to read to have it as a member on `Id` like `Id::generate()` or `Id::next()`.

**Review by MartinquaXD — APPROVED** [review]

**Comment by sunce86** [comment]
I've updated the PR according to comment from: https://github.com/cowprotocol/services/issues/2002#issuecomment-1773722203

Since we don't want to send notifications for merged settlements, then `solution_id` can be a single value.

File: crates/solvers/src/domain/solution.rs (hunk 1 above)
Comment (sunce86) [reply crates/solvers/src/domain/solution.rs]: Yes, added.

**Comment by sunce86** [comment]
I had to drop COUNTER because of issues with parallel calls of `solve` endpoint.

**Pull request merged**